newsService/src/main/java/com/kovan/app/service/NewsService.java

```diff
@@ -74,46 +73,34 @@ public List<NewsDto> getTopHeadlines() {
 
         } while (pageSize <= newsDto.getTotalResults());
 
-        AtomicReference<List<NewsDto.Article>> newsArticles = new AtomicReference<>();
-        AtomicReference<List<NewsDto.Article>> futureArticles = new AtomicReference<>();
-
         boolean isYesterdayInDb = isNewsAlreadyInDb(yesterday.toString());
         boolean isTodayInDb = isNewsAlreadyInDb(today.toString());
 
-        articles.stream()
-                .takeWhile(article -> {
-                    LocalDate date = LocalDate.parse(article.getPublishedAt(), DateTimeFormatter.ISO_DATE_TIME);
-                    logger.log(System.Logger.Level.INFO, "parsedDate = {0}", date);
-                    return !date.isBefore(yesterday);
-                })
-                .forEach(article -> {
-                    LocalDate date = LocalDate.parse(article.getPublishedAt(), DateTimeFormatter.ISO_DATE_TIME);
-                    if (date.equals(yesterday)) {
-                        logger.log(System.Logger.Level.INFO, "equal = {0}", date);
-                        newsArticles.set(createList(newsArticles.get()));
-                        newsArticles.get().add(article);
-                    } else {
-                        logger.log(System.Logger.Level.INFO, "isAfter = {0}", date);
-                        futureArticles.set(createList(futureArticles.get()));
-                        futureArticles.get().add(article);
-                    }
-                });
-
-
-        if (isNull(newsArticles.get()) && isNull(futureArticles.get()) && !isYesterdayInDb) {
+        Map<Boolean, List<NewsDto.Article>> partitionedArticles = articles.stream()
```

Comment:
At last, you found a best solution. Hardwork pays off. Keep it up!

Reply:
Thanks! Hard work really does pay off. 😊